homeassistant/components/notify/__init__.py

```diff
@@ -92,20 +96,49 @@ def notify_message(notify_service, call):
 
             title = template.render(
                 hass, call.data.get(ATTR_TITLE, ATTR_TITLE_DEFAULT))
-            target = call.data.get(ATTR_TARGET)
+            if targets.get(call.service) is not None:
+                target = targets[call.service]['target']
+            else:
+                target = call.data.get(ATTR_TARGET)
             message = template.render(hass, message)
             data = call.data.get(ATTR_DATA)
 
             notify_service.send_message(message, title=title, target=target,
                                         data=data)
 
         service_call_handler = partial(notify_message, notify_service)
-        service_name = slugify(p_config.get(CONF_NAME) or SERVICE_NOTIFY)
-        hass.services.register(DOMAIN, service_name, service_call_handler,
+        platform_name = (p_config.get(CONF_NAME) or SERVICE_NOTIFY)
+        platform_name_slug = slugify(platform_name)
+
+        if hasattr(notify_service, 'get_targets'):
```

Comment:
I like exposing the targets but this is not the correct way because it does not allow to add targets asynchronously. Then again, that would probably require a rewrite…